qt/aqt/editor.py

```diff
@@ -91,13 +91,20 @@ class Editor:
     """
 
     def __init__(
-        self, mw: AnkiQt, widget: QWidget, parentWindow: QWidget, addMode: bool = False
+        self,
+        mw: AnkiQt,
+        widget: QWidget,
+        parentWindow: QWidget,
+        addMode: bool = False,
+        *,
+        browserMode: bool = False,
     ) -> None:
         self.mw = mw
         self.widget = widget
         self.parentWindow = parentWindow
         self.note: Note | None = None
         self.addMode = addMode
+        self.browserMode = browserMode
```

Comment:
Assuming you're not planning to support add=True when browse=True, this might be better done with an enum - how about something like this?

```diff
diff --git a/qt/aqt/browser/browser.py b/qt/aqt/browser/browser.py
index 4469f2663..cf7f28275 100644
--- a/qt/aqt/browser/browser.py
+++ b/qt/aqt/browser/browser.py
@@ -18,7 +18,7 @@ from anki.notes import NoteId
 from anki.tags import MARKED_TAG
 from anki.utils import is_mac
 from aqt import AnkiQt, gui_hooks
-from aqt.editor import Editor
+from aqt.editor import Editor, EditorMode
 from aqt.exporting import ExportDialog
 from aqt.operations.card import set_card_deck, set_card_flag
 from aqt.operations.collection import redo, undo
@@ -419,7 +419,7 @@ class Browser(QMainWindow):
 
         gui_hooks.editor_did_init.append(add_preview_button)
         self.editor = aqt.editor.Editor(
-            self.mw, self.form.fieldsArea, self, browserMode=True
+            self.mw, self.form.fieldsArea, self, mode=EditorMode.BROWSE
         )
         gui_hooks.editor_did_init.remove(add_preview_button)
 
diff --git a/qt/aqt/editor.py b/qt/aqt/editor.py
index 4d1d7dba5..7ffe3c20b 100644
--- a/qt/aqt/editor.py
+++ b/qt/aqt/editor.py
@@ -13,6 +13,7 @@ import urllib.error
 import urllib.parse
 import urllib.request
 import warnings
+from enum import Enum
 from random import randrange
 from typing import Any, Callable, Match, cast
 
@@ -80,6 +81,12 @@ audio = (
 )
 
 
+class EditorMode(Enum):
+    ADD = 0
+    EDIT_CURRENT = 1
+    BROWSE = 2
+
+
 class Editor:
     """The screen that embeds an editing widget should listen for changes via
     the `operation_did_execute` hook, and call set_note() when the editor needs
@@ -95,16 +102,19 @@ class Editor:
         mw: AnkiQt,
         widget: QWidget,
         parentWindow: QWidget,
-        addMode: bool = False,
+        addMode: bool | None = None,  # legacy
         *,
-        browserMode: bool = False,
+        mode: EditorMode = EditorMode.EDIT_CURRENT,
     ) -> None:
         self.mw = mw
         self.widget = widget
         self.parentWindow = parentWindow
         self.note: Note | None = None
-        self.addMode = addMode
-        self.browserMode = browserMode
+        # legacy argument provided?
+        if addMode is not None:
+            mode = EditorMode.ADD if addMode else EditorMode.EDIT_CURRENT
+        self.addMode = mode is EditorMode.ADD
+        self.mode = mode
         self.currentField: int | None = None
         # Similar to currentField, but not set to None on a blur. May be
         # outside the bounds of the current notetype.
@@ -131,9 +141,9 @@ class Editor:
         self.web.set_bridge_command(self.onBridgeCmd, self)
         self.outerLayout.addWidget(self.web, 1)
 
-        if self.addMode:
+        if self.mode == EditorMode.ADD:
             file = "note_creator"
-        elif self.browserMode:
+        elif self.mode == EditorMode.BROWSE:
             file = "browser_editor"
         else:
             file = "reviewer_editor"
```

Reply:
This looks better indeed.